---

# Introducción a la Programación

- Curso original: [Intro to Programming](https://www.kaggle.com/learn/intro-to-programming)

| Id | Lección | URL Original | Traducción | Ejercicios |
|----|---------------------------------------------|-----------------------------------------|-------------------------------------------------------------|----------------|
| 1 | **Aritmética y Variables** | [Arithmetic and Variables](https://www.kaggle.com/code/alexisbcook/arithmetic-and-variables) | [Aritmética y Variables](https://www.kaggle.com/code/beta3logic/aritm-tica-y-variables-m01-c01) | **--Este módulo--** |
| 2 | **Funciones** | [Functions](https://www.kaggle.com/code/alexisbcook/functions) | [Funciones](https://www.kaggle.com/code/beta3logic/funciones-m02-c01) | [Ejercicios](https://www.kaggle.com/code/beta3logic/ejercicos-funciones-me02-c01) |
| 3 | **Tipos de datos** | [Data Types](https://www.kaggle.com/code/alexisbcook/data-types) | [Tipos de datos](https://www.kaggle.com/code/beta3logic/tipos-de-datos-m03-c01) | [Ejercicios](https://www.kaggle.com/code/beta3logic/ejercicios-tipos-de-datos-me03-c01) |
| 4 | **Condiciones y Sentencias condicionales**| [Conditions and Conditional Statements](https://www.kaggle.com/code/alexisbcook/conditions-and-conditional-statements) | [Condiciones y Sentencias condicionales](https://www.kaggle.com/code/beta3logic/condiciones-y-sentencias-condicionales-m04-co1) | [Ejercicios](https://www.kaggle.com/code/beta3logic/ejercicios-condiciones-me04-co1) |
| 5 | **Introducción a listas** | [Intro to Lists](https://www.kaggle.com/code/alexisbcook/intro-to-lists) | [Introducción a listas](https://www.kaggle.com/code/beta3logic/introducci-n-a-listas-m05-c01) | [Ejercicios](https://www.kaggle.com/code/beta3logic/ejercicio-introducci-n-a-listas-me05-c01) |

- Instructora y autora original: [Alexis Cook](https://www.kaggle.com/alexisbcook)

---

## ***Nota importante:*** 
Como esta es una traducción del contenido original, el certificado no será válido en Kaggle. Te recomiendo replicar el contenido de este notebook directamente en el [enlace original (Exercise)](https://www.kaggle.com/learn/intro-to-programming) para que puedas obtener tu certificado oficial sin problemas.

Este notebook comprende una serie de ejercicios para el curso [Introducción a la Programación (inglés)](https://www.kaggle.com/learn/intro-to-programming). Puedes consultar el tutorial en este [enlace](https://www.kaggle.com/alexisbcook/arithmetic-and-variables).

Este ejercicio te ayudará a empezar a ejecutar tu propio código.

# Configuración del notebook

Para comenzar, ejecuta el código en la celda siguiente.
- Empieza haciendo clic dentro de la celda de código.
- Haz clic en el triángulo (con forma de botón de "Reproducir") que aparece a la izquierda de la celda de código.
- Si tu código se ejecutó exitosamente, verás `Configuración Exitosa` como salida debajo de la celda.

En lugar de hacer clic en el triángulo, también puedes ejecutar el código presionando Shift + Enter en tu teclado. ¡Prueba esto ahora! No pasa nada malo si ejecutas el código más de una vez.


In [1]:
# Configura el ejercicio
from learntools.core import binder
binder.bind(globals())
from learntools.intro_to_programming.ex1 import *
print('Configuración Exitosa')

# Adicional para warnings
import warnings
warnings.filterwarnings("ignore")

Configuración Exitosa


El código de arriba configura el notebook para que pueda verificar tus respuestas en este ejercicio. Nunca debes modificar este código. (De lo contrario, el notebook no podrá confirmar que has completado el ejercicio exitosamente.)

Una vez que termines todas las preguntas de abajo, verás el ejercicio marcado como completo en la [página del curso](http://www.kaggle.com/learn/intro-to-programming). ¡Cuando completes todas las lecciones, recibirás un certificado de finalización del curso!

# Pregunta 1

A continuación, ejecutarás algo de código del tutorial para que veas cómo funciona por ti mismo. Ejecuta la siguiente celda de código sin cambios.


In [2]:
print("Hello, world!")

# NO ELIMINAR: Marca esta pregunta como completada
q1.check()

Hello, world!


<IPython.core.display.Javascript object>

<span style="color:#33cc33">If you see 'Hello, world!' above, You have successfully printed a message, and you're ready to move on to the next question.</span>

Acabas de ejecutar el código para imprimir `¡Hola mundo!`, que deberías ver en la salida de arriba.

La segunda línea de código (`q1.check()`) verifica tu respuesta. Nunca debes modificar este código de verificación; si lo quitas, no recibirás crédito por completar el problema.

# Pregunta 2

Ahora, imprimirás otro mensaje de tu elección. Para hacerlo, cambia `print("Escribe tu mensaje aqui!")` por un mensaje diferente. Por ejemplo, podrías cambiarlo por algo como:
- `print("¡Buenos días!")`
- `print("Estoy aprendiendo a programar :D")`

O tal vez quieras ver qué pasa si escribes algo como `print("3+4")`. ¿Retorna 7, o solo piensa en `"3+4"` como otro mensaje?

Asegúrate de que tu mensaje esté entre comillas (`"`), y que el mensaje en sí no use comillas. 
Por ejemplo, esto generará un error: `print("Ella dijo '¡buen trabajo!' y me dio un choque de manos!")` porque el mensaje contiene comillas. Si posteriomente decides tomar el curso de Python, aprenderás más sobre cómo evitar este error en la [Lección 6](https://www.kaggle.com/colinmorris/strings-and-dictionaries).

¡Siéntete libre de probar varios mensajes!

In [3]:
# ACTIVIDAD: Cambia el mensaje
print("¡Tu mensaje aquí!")

# NO ELIMINAR: Marca esta pregunta como completada
q2.check()

¡Tu mensaje aquí!


<IPython.core.display.Javascript object>

<span style="color:#33cc33">Once you have printed your own message, you're ready to move on to the next question.</span>

# Pregunta 3

Como aprendiste en el tutorial, un comentario en Python tiene un signo de numeral (`#`) al frente, lo que le dice a Python que ignore el texto después de él.

Poner un signo de numeral al frente de una línea de código hará que Python ignore ese código. Por ejemplo, esta línea sería ignorada por Python, y nada aparecería en la salida:

```python
#print(1+2)
```

Quitar este signo permitirá que el código sea ejecutado. Cuando quitamos el numeral al frente de una línea de código, se denomina **descomentar**.

En este problema, descomentarás dos líneas en la celda de código de abajo y verás la salida:
- Quita el `#` al frente de `q3.hint()`. Para evitar errores, NO quites el `#` al frente de `# Descomenta para ver la pista`.
- Después, quita el `#` al frente de `q3.solution()`.

Como en las preguntas anteriores, no cambies la línea final de código que marca tu trabajo como completado.

In [4]:
# Descomenta para obtener una pista
#q3.hint()

# Descomenta para ver la solución
#q3.solution()

# NO ELIMINAR: Verifica tu respuesta
q3.check()


<IPython.core.display.Javascript object>

<span style="color:#33cc33">Once you have printed the hint and the solution, you're ready to move on to the next question.</span>

**Traduccion:**
   *    Pista: Si alguna vez te atascas en una pregunta, es buena idea revisar la pista antes de ver la solución.

En la siguiente pregunta, y en la mayoría de los ejercicios de este curso, tendrás la opción de descomentar para ver pistas y soluciones. Una vez que te sientas cómodo sobre como realizar comentarios y revertir su acción, continúa con la siguiente pregunta.

# Pregunta 4

En el tutorial, definiste varias variables para calcular el número total de segundos en un año. Ejecuta la siguiente celda de código para hacer el cálculo aquí.


In [5]:
# Crea variables
num_anios = 4
dias_por_anio = 365 
horas_por_dia = 24
minutos_por_hora = 60
segundos_por_minuto = 60

# Calcula el numero de segundos en cuatro anios
total_segundos = segundos_por_minuto * minutos_por_hora * horas_por_dia * dias_por_anio * num_anios
print(total_segundos)

126144000


Usa la siguiente celda de código para:
- Definir una variable `nacimientos_por_minuto` y asignarle el valor 250. (En promedio nacen 250 bebes cada minuto.)
- Definir una variable `nacimientos_por_dia` que contenga el numero promedio de bebes nacidos cada dia. (Para asignar el valor de esta variable, debes usar `nacimientos_por_minuto` y algunas de las variables de la celda de código anterior.)

***Nota*** : Los nombres de las variables deben ser en ingles `births_per_min` y `births_per_day` para que el sistema sea capaz de reconocer el acierto.

¡Recuerda que siempre puedes obtener una pista si la necesitas!


In [6]:
# ACTIVIDAD: Asigna el valor de la variable nacimientos_por_minuto (births_per_min)
births_per_min = 000

# ACTIVIDAD: Asigna el valor de la variable nacimientos_por_dia (births_per_day)
births_per_day = 000

# NO ELIMINAR: Verifica tu respuesta
q4.check()

<IPython.core.display.Javascript object>

<span style="color:#cc3333">Incorrect:</span> Incorrect value for `births_per_min`: `0`

In [7]:
# Descomenta para obtener una pista
#q4.hint()

# Descomenta para ver la solución
#q4.solution()

# 🌶️ Pregunta 5

(¡Las preguntas marcadas con 🌶️ serán un poco más desafiantes que las demás! Recuerda que siempre puedes obtener una pista o ver la solución.)

La [competencia del Titanic](https://www.kaggle.com/c/titanic) es la competencia de ciencia de datos más famosa de Kaggle. En esta competencia, los participantes deben construir un modelo de machine learning que prediga si los pasajeros sobrevivieron o no al naufragio del Titanic, basado en información como edad, sexo, tamaño de la familia y número de boleto.

Ejecuta la siguiente celda de código sin cambios para cargar y previsualizar los datos del Titanic.

No te preocupes por los detalles del código por ahora: el resultado final es que todos los datos del Titanic se han cargado en una variable llamada `titanic_data`. (Para aprender a escribir este código tú mismo, puedes tomar el [curso de Python](https://www.kaggle.com/learn/python) y luego el [curso de Pandas](https://www.kaggle.com/learn/pandas).)


In [8]:
# Carga los datos de la competencia del Titanic
import pandas as pd
titanic_data = pd.read_csv("../input/titanic/train.csv")

# Muestra las primeras cinco filas de los datos
titanic_data.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

Los datos tienen una fila diferente para cada pasajero.

La siguiente celda de código define e imprime los valores de tres variables:
- `total` = número total de pasajeros que abordaron el barco
- `sobrevivieron` = número de pasajeros que sobrevivieron al naufragio
- `menores` = número de pasajeros menores de 18 años

Ejecuta la celda de código sin cambios. (No te preocupes por los detalles de cómo se calculan estas variables por ahora. Puedes aprender más sobre cómo calcular estos valores en el [curso de Pandas](https://www.kaggle.com/learn/pandas).)


In [9]:
# Numero de pasajeros total
total = len(titanic_data)
print(total)

# Numero de pasajeros que sobrevivieron
sobrevivieron = (titanic_data.Survived == 1).sum()
print(sobrevivieron)

# Numero de pasajeros menores de 18
menores = (titanic_data.Age < 18).sum()
print(menores)

891
342
113


Entonces,
- `total = 891` (había 891 pasajeros a bordo del Titanic),
- `sobrevivieron = 342` (342 pasajeros sobrevivieron), y
- `menores = 113` (113 pasajeros eran menores de 18 años).

En la celda de código de abajo, reemplaza las líneas con valores `000`  con código para calcular los valores de dos variables más:
- `fraccion_sobrevivientes` debe ser la fracción de pasajeros que sobrevivieron al desastre del Titanic.
- `fraccion_menores` debe ser la fracción de pasajeros que eran menores (menores de 18 años).

Para cada variable, tu respuesta debe ser un número entre 0 y 1.

***Nota*** : Los nombres de las variables deben ser en ingles `survived_fraction` y `minors_fraction` para que el sistema sea capaz de reconocer el acierto.


Si necesitas una pista o quieres ver la solución, puedes ir a la siguiente celda de código y descomentar las líneas correspondientes (`q5.hint()` y `q5.solution()`).


In [10]:
# ACTIVIDAD: Completa el valor de la variable fraccion_sobrevivientes
survived_fraction = 000

# Imprime el valor de la variable
print(survived_fraction)

# ACTIVIDAD: Completa el valor de la variable fraccion_menores
minors_fraction = 00

# Imprime el valor de la variable
print(minors_fraction)

# NO ELIMINAR: Verifica tu respuesta
q5.check()


0
0


<IPython.core.display.Javascript object>

<span style="color:#cc3333">Incorrect:</span> Incorrect value for `survived_fraction`: `0`

In [11]:
# Uncomment to receive a hint
#q5.hint()

# Uncomment to view the solution
#q5.solution()

# Ejercicio Extra

Como un **opcional** siguiente paso, te animo a hacer una submission a la competencia del Titanic.  
> Para hacerlo, sigue las instrucciones en el **[notebook aquí](https://www.kaggle.com/alexisbcook/titanic-tutorial)**.  

El notebook es amigable para principiantes y no asume que tienes experiencia con programación. Al mismo tiempo, aprenderás cómo crear tus propios notebooks en Kaggle, ¡y te dará una gran idea de lo que podrás hacer si sigues aprendiendo ciencia de datos!

# Sigue adelante

¡Felicidades! Acabas de terminar el primer ejercicio del curso Intro a la Programación!  

Ahora estás listo para aprender cómo **[organizar tu código con funciones](https://www.kaggle.com/alexisbcook/functions)**.


---




*¿Tienes preguntas o comentarios? Visita el [foro de discusión (inglés)](https://www.kaggle.com/learn/intro-to-programming/discussion) del curso para charlar con otros estudiantes.*

```
  _  __    _    ____  ____ _     _____ 
 | |/ /   / \  / ___|/ ___| |   | ____|
 | ' /   / _ \| |  _| |  _| |   |  _|  
 | . \  / ___ \ |_| | |_| | |___| |___ 
 |_|\_\/_/   \_\____|\____|_____|_____|
                                       
```